mantidproject/vesuvio
# Add ability to globally fit a shared parameter

**Description of work:**

This PR contains work that @jess-farmer has done in order to enable the global fitting of shared parameters.

In addition to this, I have added a system test and slightly refactored some of the code.

**To test:**

1. Ensure system tests pass (note that the new global fitting takes quite a while to run).
2. Ensure unit tests pass
3. Load the algorithms into `mantid`, ensure the new drop down menus to select `SharedParameterFitType` work.

This PR leaves a number of issues: #59

"Code quality" is not a priority, this code will be refactored and improved where necessary in a subsequent PR (see above issue).

This PR resolves: #60
